### PR TITLE
Sensei tweaks, Training Feature

### PR DESCRIPTION
**Pull request by aumu1031**
- Sensei are considered specialized in their bloodline
- Fix cast speed specialization display
- Sensei set to inactive instead of deleted to preserve data

### CODE REVIEW

File: classes/SenseiManager.php
Comment (levimeahan) [inline classes/SenseiManager.php:103]: We should make a `SenseiInfoDto` for this, much easier to work with outside this function

File: classes/SenseiManager.php
Comment (levimeahan) [inline classes/SenseiManager.php:204]:
```suggestion
        $system->db->query(
            "UPDATE `sensei` SET `enable_lessons` = '{$enable_lessons}' WHERE `sensei_id` = '{$sensei_id}'"
        );
        
        return $system->db->last_affected_rows > 0;
```

File: classes/SenseiManager.php
Comment (levimeahan) [inline classes/SenseiManager.php:420]: We should not redeclare these values here or it will get out of date if we tweak training page, can we pull this to a method on the `User` class? Or create a shell of a `TrainingManager` class (in future to be used via API, but for now just to hold this "how are trainings setup" logic)

File: classes/SenseiManager.php
Comment (levimeahan) [inline classes/SenseiManager.php:440]: same comment, reference from a single source of truth instead of redeclaring here

File: classes/SenseiManager.php
Comment (levimeahan) [inline classes/SenseiManager.php:465]: I generally lean towards being verbose and explicit but I wonder if this provides that much benefit. If anything maybe we add a comment here, but IMO the logic is straightforward enough to read
```suggestion
        return $system->db->last_affected_rows > 0;
```

File: pages/premium.php
Comment (levimeahan) [inline pages/premium.php:113]:
```suggestion
                throw new RuntimeException("You must leave your team before resetting!");
```
This looks like bad merge conflict, these should all be RuntimeException now

File: pages/training.php
Comment (levimeahan) [inline pages/training.php:147]: Maybe we could simplify this to a `SenseiManager::getTrainingBoostForTrainType($player->sensei_id, $train_type, $system)`

which in most cases returns 0 or null, but in these two cases returns `$sensei_boost['boost_primary']`

File: pages/villageHQ.php
Comment (levimeahan) [inline pages/villageHQ.php:281]: More merge conflicts here
```suggestion
                    throw new RuntimeException('You do not meet the requirements!');
```

File: pages/training.php
Comment (aumu1031) [reply pages/training.php]: Modified in commit.

File: db/migrations/20230620062917_sensei_lesson_migration.php
Comment (aumu1031) [reply db/migrations/20230620062917_sensei_lesson_migration.php]: Fixed migration in commit, had other issues.

File: classes/SenseiManager.php
Comment (aumu1031) [reply classes/SenseiManager.php:117]: Agreed. I'm thinking this is a to-do as part of cleaning up the sensei functionality and converting to API.

File: pages/training.php
Comment (aumu1031) [reply pages/training.php]: Changed in latest commit.

File: classes/SenseiManager.php
Comment (aumu1031) [reply classes/SenseiManager.php:431]: As part of the hotbar implementation I assume we're moving training, ramen and special missions into APIs? 
So as part of that the training logic on the Training/Sensei/Premium pages will be unified.

File: pages/villageHQ.php
Comment (levimeahan) [inline pages/villageHQ.php:589]: Tbh I would really prefer we centralize these in the PR that they are introduced (currently, nowhere outside of `training.php` attempts to define these values) rather than introducing them and then cleaning up later. Would be okay if it's in total shell of a `TrainingManager` class which is only used (for now) for exposing these values to wherever needs them.

But I am okay with that being an immediate fast follow

**Review by levimeahan — APPROVED** [review]
Approving to unblock, but two pressing comments need to be addressed ASAP (fixing exceptions from the merge conflict is most important)

File: classes/SenseiManager.php
Comment (aumu1031) [reply classes/SenseiManager.php]: Moved to TrainingManager

File: classes/SenseiManager.php
Comment (aumu1031) [reply classes/SenseiManager.php]: Moved to TrainingManager

File: pages/villageHQ.php
Comment (aumu1031) [reply pages/villageHQ.php]: Updated Exceptions -> RuntimeException

File: pages/villageHQ.php
Comment (aumu1031) [reply pages/villageHQ.php]: Moved to TrainingManager

**Pull request merged**
